Aplib.Core/Collections/ExposedQueue.cs

```diff
@@ -45,30 +46,31 @@ public ExposedQueue(int size)
         }
 
         /// <summary>
-        /// Initializes a new instance of the <see cref="ExposedQueue{T}"/> class
+        /// Initializes a new instance of the <see cref="ExposedQueue{T}" /> class
         /// with an array to use as basis for the queue.
-        /// By default, assumes the array is filled. 
+        /// By default, assumes the array is filled.
         /// </summary>
         /// <param name="array">An array to use as the circular array.</param>
-        /// <param name="count">The number of actual elements in the array.</param>
+        /// <param name="maxCount">The number of actual elements in the array.</param>
         /// <remarks>
         /// The MaxCount of the queue will be set to the length of the array.
         /// If the array is not fully filled, the Count should be specified.
         /// </remarks>
-        public ExposedQueue(T[] array, int count)
+        public ExposedQueue(T[] array, int maxCount)
         {
-            if (count > array.Length)
-                throw new ArgumentOutOfRangeException(nameof(count), "Count cannot exceed the length of the array.");
-            if (count < 0)
-                throw new ArgumentOutOfRangeException(nameof(count), "Count cannot be negative.");
+            if (maxCount < 0) throw new ArgumentOutOfRangeException(nameof(maxCount), "Count cannot be negative.");
 
-            MaxCount = array.Length;
-            _array = array;
-            _head = MaxCount - 1;
-            Count = count;
+            MaxCount = maxCount;
+            _array = new T[maxCount];
+
+            if (array.Length > maxCount) array = array[..maxCount];
+
+            array.CopyTo(_array, 0);
+            Count = array.Length;
```

Comment:
If array.Length > maxCount, the array is copied twice. Can this be done more efficiently?
For example 
```suggestion
            Count = Math.Min(array.Length, MaxCount);
            
            _array = new T[maxCount];
            Array.Copy(array, _array, Count);
```